abi.go

```diff
@@ -99,6 +112,25 @@ func newMapABIFromFd(fd *bpfFD) (*MapABI, error) {
 	}, nil
 }
 
+func newMapABIFromProc(fd *bpfFD) (*MapABI, error) {
+	var abi MapABI
+	err := scanFdInfo(fd, map[string]interface{}{
+		"map_type":    &abi.Type,
+		"key_size":    &abi.KeySize,
+		"value_size":  &abi.ValueSize,
+		"max_entries": &abi.MaxEntries,
+	})
```

Comment:
The ABI doesn't have flags since I've not had a use case for them. I'm happy to add them if they are useful to you.

Note that `MapABI` and `MapSpec` are different though, so you'd not be able to use this to create a new map of a certain type.